### PR TITLE
[ethash] chainspec validate `ecip1017EraRounds` non-zero

### PR DESCRIPTION
**Pull request by niklasad1**
See https://github.com/paritytech/parity-ethereum/issues/11119 for background information.

This PR does the following:

1. Modifies chain spec validation to require ecip1017EraRounds to be non-zero. For a more human-friendly error message. (Could maybe be bigger than 1 instead, however, it is technically a breaking change but it makes parity useless)
2. Document `fn ecip1017_eras_block_reward` and add explicit panic for `hidden` panics that may occur by `divide by zero`

We could maybe just return `zero era rounds` when  `ecip1017EraRounds < 2`

Thoughts?

### CODE REVIEW

File: ethcore/engines/ethash/src/lib.rs [hunk 1]
```diff
@@ -464,7 +464,18 @@ impl Ethash {
...
+/// Calculates the number of eras and reward
+///
+/// # Panics
+///
+/// This function will panic if `era_rounds` is less than `2`
+fn ecip1017_eras_block_reward(era_rounds: u64, mut reward: U256, block_number: u64) -> (u64, U256) {
+	// NOTE(niklasad1): all numbers is divisible by 1, it will cause the if below
+	// to succeed except the first block. Thus, `era_rounds - 1 == 0` and cause `divide by zero`
+	//
+	// TODO(niklasad): We could set `eras` to zero if this occurs, but I'm not sure of the implications
```
Comment (niklasad1) [inline ethcore/engines/ethash/src/lib.rs]: thoughts?

File: ethcore/engines/ethash/src/lib.rs [hunk 2]
```diff
@@ -464,7 +464,18 @@ impl Ethash {
...
-fn ecip1017_eras_block_reward(era_rounds: u64, mut reward: U256, block_number:u64) -> (u64, U256) {
+
+/// Calculates the number of eras and reward
+///
+/// # Panics
+///
+/// This function will panic if `era_rounds` is less than `2`
+fn ecip1017_eras_block_reward(era_rounds: u64, mut reward: U256, block_number: u64) -> (u64, U256) {
+	// NOTE(niklasad1): all numbers is divisible by 1, it will cause the if below
+	// to succeed except the first block. Thus, `era_rounds - 1 == 0` and cause `divide by zero`
```
Comment (niklasad1) [inline ethcore/engines/ethash/src/lib.rs]:
```suggestion
	// to succeed except for the first block. Thus, `era_rounds - 1 == 0` and cause `divide by zero`
```

File: ethcore/engines/ethash/src/lib.rs (hunk 1 above)
Comment (ordian) [reply ethcore/engines/ethash/src/lib.rs]: I think it's fine to assume blocks in one rounds is at least 2 (it's set to 5_000_000 for ETC if I'm not mistaken).

Comment (ordian) [reply ethcore/engines/ethash/src/lib.rs]: Let's remove the TODO and we're good to merge.

Comment (niklasad1) [reply ethcore/engines/ethash/src/lib.rs]: the note too?

Comment (ordian) [reply ethcore/engines/ethash/src/lib.rs]: the note is good for future readers

**Pull request merged**